### PR TITLE
[Merged by Bors] - chore(MeasureTheory.Constructions.BorelSpace.Basic): split file

### PR DESCRIPTION
**Pull request by kkytola**
The file `MeasureTheory.Constructions.BorelSpace.Basic` was approximately 2500 lines long. There is somewhat independent content related to order topologies, metrics, and reals, ennreals, and nnreals. This PR proposes to spilt the file according to those boundaries: `Basic` < `Order` < `Real` < `Metric`. (Short module docstring summaries are added to the new files.)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by grunweg** [comment]
Great to see more file splits, thank you for doing this! Friendly reminder that you can remove the style exception for this file. :-)

TL;DR: remove this line:

https://github.com/leanprover-community/mathlib4/blob/f1d422f45ad94f00f9b1ed78b9e28f0fb3c26ab8/scripts/style-exceptions.txt#L72

**Comment by kkytola** [comment]
> Great to see more file splits, thank you for doing this! Friendly reminder that you can remove the style exception for this file. :-)
> 
> TL;DR: you can remove this line now:
> 
> https://github.com/leanprover-community/mathlib4/blob/f1d422f45ad94f00f9b1ed78b9e28f0fb3c26ab8/scripts/style-exceptions.txt#L72

This is my first attempt of a large file split, so I don't yet guarantee that I manage to pull this out --- the PR is a "work in progress". But I got started with this attempt, since I recently added something in this long file and then felt a bit guilty.

File: Mathlib/MeasureTheory/Constructions/BorelSpace/Order.lean [hunk 1]
```diff
@@ -0,0 +1,1032 @@
+/-
+Copyright (c) 2017 Johannes Hölzl. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johannes Hölzl, Yury Kudryashov, Kexing Ying
```
Comment (kkytola) [inline Mathlib/MeasureTheory/Constructions/BorelSpace/Order.lean:4]: I kept otherwise the original authors and copyright, but I added @JasonKYi to this part of the split file, because from, e.g., [this mathlib3 PR 9396](https://github.com/leanprover-community/mathlib/pull/9396) it seems that a fair bit of the material in this part was contributed by them.

File: Mathlib/MeasureTheory/Constructions/BorelSpace/Order.lean [hunk 2]
```diff
@@ -0,0 +1,1032 @@
...
+import Mathlib.MeasureTheory.Constructions.BorelSpace.Basic
+
+#align_import measure_theory.constructions.borel_space.basic from "leanprover-community/mathlib"@"9f55d0d4363ae59948c33864cbc52e0b12e0e8ce"
+
+/-!
+# Borel sigma algebras on spaces with orders
+
+## Main statements
+
+* `borel_eq_generateFrom_Ixx` (where Ixx is one of {Iio, Ioi, Iic, Ici, Ico, Ioc}):
```
Comment (kkytola) [inline Mathlib/MeasureTheory/Constructions/BorelSpace/Order.lean:19]: I added rough descriptions of "main statements", based on a rough judgement.

File: Mathlib/MeasureTheory/Constructions/BorelSpace/Real.lean [hunk 3]
```diff
@@ -0,0 +1,551 @@
...
+Authors: Johannes Hölzl, Yury Kudryashov, Rémy Degenne
+-/
+import Mathlib.MeasureTheory.Constructions.BorelSpace.Order
+
+#align_import measure_theory.constructions.borel_space.basic from "leanprover-community/mathlib"@"9f55d0d4363ae59948c33864cbc52e0b12e0e8ce"
+
+/-!
+# Borel (measurable) spaces ℝ, ℝ≥0, ℝ≥0∞
+
+## Main statements
```
Comment (kkytola) [inline Mathlib/MeasureTheory/Constructions/BorelSpace/Real.lean:13]: I added rough descriptions of "main statements", based on a rough judgement.

**Comment by adomani** [comment]
As a test of the `no_lost_declarations` script, this is the output:

* `-` `theorem measure_eq_measure_preimage_add_measure_tsum_Ico_zpow [MeasurableSpace α] (μ : Measure α)`
* `+` `theorem measure_eq_measure_preimage_add_measure_tsum_Ico_zpow {α : Type*} [MeasurableSpace α]`

---

* 1  added declarations
* 1  removed declarations
* 188  paired declarations

---

**Comment by kkytola** [comment]
> As a test of the `no_lost_declarations` script, this is the output:
>     * `-` `theorem measure_eq_measure_preimage_add_measure_tsum_Ico_zpow [MeasurableSpace α] (μ : Measure α)`
>     * `+` `theorem measure_eq_measure_preimage_add_measure_tsum_Ico_zpow {α : Type*} [MeasurableSpace α]`
> 
>     * 1  added declarations
>     * 1  removed declarations
>     * 188  paired declarations

Thanks! Where can I find and run the `no_lost_declarations` script? I looked in the `/scripts` folder and could not see it there, and I searched Zulip and found nothing...

The result looks pretty reassuring, right? Except I'm not totally sure why the one removed-then-added declaration difference. The type signature is not displayed completely, but there seems to be a small difference at the start... However, when I compare my VS Code hover with [Loogle result](https://loogle.lean-lang.org/?q=measure_eq_measure_preimage_add_measure_tsum_Ico_zpow), these look identical modulo a universe variable name. So I'll be reassured unless someone tells me otherwise :smile:. Thanks a lot for running this and letting me know of the existence of such a script!

**Comment by adomani** [comment]
@kkytola, I only now saw your message, since I did not get/lost the notification.

The script does not exist on master: it is only present in #12515.

The reason for the difference is that the script is text-based, so it is highlighting that the two lines are not *identical*: indeed,
* the first contains `(μ : Measure α)`, while the second does not;
* the second contains `{α : Type*}`, while the first does not.

So, as long as these changes are ok (which presumably they are, since there is likely a `variable` and a line-break difference), the script essentially is telling you that no declarations were lost!

If you want to run the script locally, you can check it out from the PR and then run it as follows:
```bash
git checkout origin/adomani/no_lost_declarations scripts/no_lost_declarations.sh
./scripts/no_lost_declarations.sh
```

Note that the code above will add the file `scripts/no_lost_declarations.sh` so you would have to make sure that you do not PR that file as well!

**Comment by RemyDegenne** [comment]
Please merge master to fix the conflict. Then let's merge that PR soon.

**Comment by RemyDegenne** [comment]
Thanks!
bors r+
